chain-impl-mockchain/tests/ledger/ledger_tests.rs

```diff
@@ -0,0 +1,232 @@
+use crate::common::accounts;
+use crate::common::ledger;
+use crate::common::ledger::ConfigBuilder;
+use crate::common::tx_builder::TransactionBuilder;
+use chain_addr::Discrimination;
+
+use chain_impl_mockchain::account::Identifier;
+use chain_impl_mockchain::account::SpendingCounter;
+
+use chain_impl_mockchain::ledger::Error::NotEnoughSignatures;
+use chain_impl_mockchain::transaction::*;
+use chain_impl_mockchain::value::*;
+
+macro_rules! assert_err {
```

Comment:
Could just use `.expect_err()` and examine the returned error with `assert_eq!`.